cli/src/cli_util.rs

```diff
@@ -3916,6 +3916,10 @@ impl<'a> CliRunner<'a> {
             process_global_args_fn(ui, &matches)?;
         }
         config_env.set_command_name(command_name(&matches));
+        // Set hostname for --when.hostnames conditional config scopes.
+        if let Ok(hostname) = raw_config.as_ref().get::<String>("operation.hostname") {
```

Comment:
Better to not use config value to resolve config conditionals because that means the resolved layer may override the config value that was previously used. Just use `whoami::fallible::hostname()` instead?

Reply:
Agree.